server/routes/matching/match_jobs.py

```diff
@@ -46,94 +46,118 @@ def match_jobs():
                 id = data.get("user_id")
 
                 job_match = []
+                cand_skills = []
+                cand_levels = []
+                cand_values = []
+                cand_soft_skills = []
+
                 cand_json = {"user_id": id}
 
                 candidate = requests.post(
                     f"{domain_name}/api/get_candidate_by_id", json=cand_json
                 )
-                print("CANDIDATE", candidate.json())
                 if candidate.status_code == 200:
-                    cand_skills = [
-                        skill["skill_name"]
-                        for skill in candidate.json()["candidates"]["skills"]
-                    ]
-                    cand_levels = [
-                        skill["skill_level"]
-                        for skill in candidate.json()["candidates"]["skills"]
-                    ]
-                    cand_values = candidate.json()["candidates"]["values"]
-                    cand_soft_skills = candidate.json()["candidates"]["soft_skills"]
+                    if len(candidate.json()["candidates"]["skills"]) > 0:
+                        candidate_skills = [
+                            skill["skill_name"]
+                            for skill in candidate.json()["candidates"]["skills"]
+                        ]
+                        candidate_skills.append(cand_skills)
+                        candidate_levels = [
+                            skill["skill_level"]
+                            for skill in candidate.json()["candidates"]["skills"]
+                        ]
+                        candidate_levels.append(cand_levels)
+
+                    if len(candidate.json()["candidates"]["values"]) > 0:
+                        candidate_values = candidate.json()["candidates"]["values"]
+                        cand_values.append(candidate_values)
+
+                    if len(candidate.json()["candidates"]["soft_skills"]) > 0:
+                        candidate_soft_skills = candidate.json()["candidates"][
+                            "soft_skills"
+                        ]
+                        cand_soft_skills.append(candidate_soft_skills)
```

Comment:
I don't these checks are needed:

These keys are always present in the candidate response. The only thing is that they might not have associated values:

Therefore they could be have values or be null / None. 

```python
if None:
  print(a)
```
This will not print a. The previous code snippet was adequte.

Reply:
hey isak, here I'm checking if the array it's not **empy**, I'm not checking if candidate is None. What happens is that some candidates may have either the **skills**, **soft_skills** or **values** to be either **empty** (len = 0) or **None** and in that case the call was failing and not updating the candidate match_jobs array.